src/tarball-file-system.ts

```diff
@@ -36,7 +36,11 @@ export class TarballFileSystem implements FileSystem {
 
   async readFile(filePath: string): Promise<string> {
     const {files} = await this.#getUnpackResult();
-    const fullPath = path.join(await this.getRootDir(), filePath);
+
+    const cleanPath = filePath.startsWith('/') ? filePath.slice(1) : filePath;
```

Comment:
I would only suggest you to rename this var for clarity 😁

```suggestion
    const normalizedPath = filePath.startsWith('/') ? filePath.slice(1) : filePath;
```